snmp/tests/test_check.py

```diff
@@ -488,18 +488,19 @@ def test_forcedtype_metric(aggregator):
     aggregator.all_metrics_asserted()
 
 
-def test_invalid_forcedtype_metric(aggregator):
+def test_invalid_forcedtype_metric(aggregator, caplog):
     """
-    If a forced type is invalid a warning should be issued + a service check
-    should be available
+    If a forced type is invalid a warning should be issued
+    but the check should continue processing the remaining metrics.
     """
     instance = common.generate_instance_config(common.INVALID_FORCED_METRICS)
     check = common.create_check(instance)
 
     check.check(instance)
 
-    # Test service check
-    aggregator.assert_service_check("snmp.can_check", status=SnmpCheck.WARNING, tags=common.CHECK_TAGS, at_least=1)
+    aggregator.assert_service_check("snmp.can_check", status=SnmpCheck.OK, tags=common.CHECK_TAGS, at_least=1)
```

Comment:
Should we keep the warning level in this case?

Reply:
The warning service check is a consequence of the `ConfigurationError` that was raised before. 

As indicated in the desc, I think:
>  Let's not raise ConfigurationError so that the check continue to work even if there is an issue with one of the metrics

Reply:
@AlexandreYang I agree the check should continue running if there is an issue with one metric.
What I was thinking was to specify that an issue occurred after submitting all the metrics.

Something like
```python
try:
    self._do_submit_metric(name, snmp_value, forced_type, tags, options)
except Exception as e:
    msg = ...
    self.log.warning(msg)
    self.log.debug(msg, exc_info=True)
    return False
return True

[...]

if not self.submit_metric(...):
    self.service_check("snmp.can_check", SnmpCheck.WARNING)         
```
So the check still submits metric but report the issue in the service check at the end of the run. I don't know if it wouldn't be too loud however. WDYT?

Reply:
That's possible, but I'm not sure it worth it since it will add some complexity to the integration and I'm not sure if that really brings value to user. They can already monitor the Agent logs for warning messages.

Also, I think more important than above, the try/except is now more generic and we might send more WARNING service checks than before, which might impact customers monitors.

Removing `ConfigurationError` also changes service checks, but it's more an edge case I think in comparison.

Reply:
Sure, just wanted a clarification here, there might be more warning with this try/except indeed, and the logging seems enough.